### PR TITLE
refactor: convert function comments to doxygen format

### PR DESCRIPTION
**Pull request by dundargoc**
Ramping up the refactoring rate, this'll take too long otherwise :)

### CODE REVIEW

**Comment by bfredl** [comment]
This a little big. I'd suggest break out `eval.c` , `eval/funcs.c` and `ex_docmd.c` into separate PR:s. The rest can stay here.

File: src/nvim/ex_cmds2.c [hunk 1]
```diff
@@ -492,8 +492,8 @@ void autowrite_all(void)
   }
 }
 
-/// Return true if buffer was changed and cannot be abandoned.
-/// For flags use the CCGD_ values.
+/// @return  true if buffer was changed and cannot be abandoned.
+///          For flags use the CCGD_ values.
```
Comment (zeertzjq) [inline src/nvim/ex_cmds2.c:496]:
```suggestion
/// For flags use the CCGD_ values.
```

**Comment by zeertzjq** [comment]
There are several mistakes in `ex_getln.c`. I want to block this after #17726.

**Comment by bfredl** [comment]
In the meanwhile, I'd suggest breaking out `ex_eval.c` and `fold.c` to separate PR:s :)

**Comment by dundargoc** [comment]
Broke out ex_eval, fold and ex_getln.

File: src/nvim/extmark.c [hunk 2]
```diff
@@ -287,12 +288,12 @@ bool extmark_clear(buf_T *buf, uint32_t ns_id, int l_row, colnr_T l_col, int u_r
...
 }
 
-// Returns the position of marks between a range,
-// marks found at the start or end index will be included,
-// if upper_lnum or upper_col are negative the buffer
-// will be searched to the start, or end
-// dir can be set to control the order of the array
-// amount = amount of marks to find or -1 for all
+/// @return  the position of marks between a range,
+/// marks found at the start or end index will be included,
```
Comment (bfredl) [inline src/nvim/extmark.c:292]: indentation

File: src/nvim/fileio.c [hunk 3]
```diff
@@ -4860,11 +4851,10 @@ int check_timestamps(int focus)
...
 
-/*
- * Move all the lines from buffer "frombuf" to buffer "tobuf".
- * Return OK or FAIL.  When FAIL "tobuf" is incomplete and/or "frombuf" is not
- * empty.
- */
+/// Move all the lines from buffer "frombuf" to buffer "tobuf".
+///
+/// @return  OK or FAIL.  When FAIL "tobuf" is incomplete and/or "frombuf" is not
+///                       empty.
```
Comment (bfredl) [inline src/nvim/fileio.c:4857]: indentation

**Review by bfredl — APPROVED** [review]

**Pull request merged**